jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/javacore/parser/j9/section/memory/MemorySectionParser.java

```diff
@@ -59,16 +66,50 @@ protected void topLevelRule() throws ParserException {
 	 */
 	private void memInfo() throws ParserException {
 
-		IAttributeValueMap results = null;
+		IAttributeValueMap results;
+
+		while ((results = processTagLineOptional(T_1STHEAPTYPE)) != null) {
+			// The heap type
+			String segName = results.getTokenValue(MEMORY_SEGMENT_NAME);
+			IAttributeValueMap resultsSpace;
+			while ((resultsSpace = processTagLineOptional(T_1STHEAPSPACE)) != null) {
+				long spaceid = resultsSpace.getLongValue(MEMORY_SEGMENT_ID);
+				String spacetype = resultsSpace.getTokenValue(MEMORY_SEGMENT_TYPE).trim();
+				String spaceName;
+				if (spaceid != IBuilderData.NOT_AVAILABLE) {
+					spaceName = segName + " " + spacetype + " 0x" + Long.toHexString(spaceid);
+				} else {
+					spaceName = segName + " " + spacetype;
+				}
+				JavaHeap heap = fJavaRuntimeBuilder.addJavaHeap(spaceName);
```

Comment:
`fJavaRuntimeBuilder` might be `null` here.

Reply:
Done